src/overlays/actors/ovl_En_Honotrap/z_en_honotrap.c

```diff
@@ -43,127 +85,729 @@ ActorInit En_Honotrap_InitVars = {
+        this->actor.child = NULL;
+    }
+    this->actionFunc(this, play);
+    if (this->collider.tris.base.acFlags & AC_HIT) {
+        EffectSsBomb2_SpawnLayered(play, &this->actor.world.pos, &sVelocity, &sAccel, 15, 8);
+        Actor_Kill(&this->actor);
+        return;
+    }
+    if (this->eyeState < HONOTRAP_EYE_SHUT) {
+        this->collider.tris.base.acFlags &= ~AC_HIT;
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.tris.base);
+    }
+}
+
+void EnHonotrap_UpdateFlame(Actor* thisx, PlayState* play) {
+    EnHonotrap* this = (EnHonotrap*)thisx;
+
+    if (this->timer > 0) {
+        this->timer--;
+    }
+    this->colChkFlags = 0;
+    this->bobPhase += 0x640;
+    if (this->actor.params != HONOTRAP_TYPE_FLAME_GROUP) {
+        this->actor.shape.yOffset = Math_SinS(this->bobPhase) * 1000.0f + 600.0f;
+    }
+    Actor_SetFocus(&this->actor, 5.0f);
+    Actor_PlaySfx(&this->actor, NA_SE_EV_BURN_OUT - SFX_FLAG);
+    this->actionFunc(this, play);
+    this->flameScroll -= 20;
+    this->flameScroll &= 0x1FF;
+}
+
+void EnHonotrap_UpdateFlameGroup(Actor* thisx, PlayState* play) {
+    EnHonotrap* this = (EnHonotrap*)thisx;
+    Actor* parent = this->actor.parent;
+
+    if ((parent != NULL) && (parent->update == NULL)) {
+        this->actor.parent = NULL;
+    }
+    if (this->timer > 0) {
+        this->timer--;
+    }
+    this->bobPhase += 0x640;
+    Actor_PlaySfx(&this->actor, NA_SE_EV_BURN_OUT - SFX_FLAG);
+    this->actionFunc(this, play);
+}
+
+void EnHonotrap_Draw(Actor* thisx, PlayState* play) {
+    EnHonotrap* this = (EnHonotrap*)thisx;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    Gfx_SetupDL25_Opa(play->state.gfxCtx);
+    gSPSegment(POLY_OPA_DISP++, 0x08, sSilverEyeTextures[this->eyeState]);
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, &gEyeSwitchSilverDL);
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+void EnHonotrap_DrawFlame(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnHonotrap* this = (EnHonotrap*)thisx;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    Gfx_SetupDL25_Xlu(play->state.gfxCtx);
+    gSPSegment(POLY_XLU_DISP++, 0x08,
+               Gfx_TwoTexScroll(play->state.gfxCtx, 0, 0, 0, 32, 64, 1, 0, this->flameScroll, 32, 128));
+    gDPSetPrimColor(POLY_XLU_DISP++, 0x80, 0x80, 255, 200, 0, 255);
+    gDPSetEnvColor(POLY_XLU_DISP++, 255, 0, 0, 0);
+    Matrix_RotateYS(Camera_GetCamDirYaw(GET_ACTIVE_CAM(play)) + 0x8000, MTXMODE_APPLY);
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_XLU_DISP++, gEffFire1DL);
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+void EnHonotrap_DrawFlameGroup(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnHonotrap* this;
+    EnHonotrapFlameElement* flameElem;
+    EnHonotrapFlameGroup* flameGroup;
```

Comment:
```suggestion
    EnHonotrap* this = (EnHonotrap*)thisx ;
    EnHonotrapFlameElement* flameElem;
    EnHonotrapFlameGroup* flameGroup = &this->flameGroup;
```
And removing the assignments down below work for me.